QuizApp/QuizApp/UI/Screens/Leaderboard/LeaderboardViewController.swift

```diff
@@ -0,0 +1,167 @@
+
+        let image = UIImage(systemName: "xmark", withConfiguration: UIImage.SymbolConfiguration(weight: .bold))
+        let closeButton = UIBarButtonItem(image: image, style: .done, target: self, action: #selector(pressedClose))
+        closeButton.tintColor = .white
+        navigationItem.rightBarButtonItem = closeButton
+
+        playerLabel.text = "Player"
+        playerLabel.textColor = .white
+        playerLabel.font = UIFont(name: "SourceSansPro-Regular", size: 16)
+
+        pointsLabel.text = "Points"
+        pointsLabel.textColor = .white
+        pointsLabel.font = UIFont(name: "SourceSansPro-Regular", size: 16)
+
+        let border = UIView()
+        let onePixel = 1 / UIScreen.main.scale
+
+        border.frame = CGRect(x: 0, y: 0, width: view.frame.size.width, height: onePixel)
+        border.backgroundColor = .white
+        border.autoresizingMask = [.flexibleWidth, .flexibleBottomMargin]
+
+        tableView.addSubview(border)
+        tableView.backgroundColor = .clear
+        tableView.allowsSelection = false
+        tableView.register(LeaderboardCell.self, forCellReuseIdentifier: LeaderboardCell.reuseIdentifier)
+        tableView.dataSource = self
+        tableView.separatorColor = .white
+        tableView.separatorInset = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
+    }
+
+    @objc
+    private func pressedClose() {
+        viewModel.pressedClose()
+    }
+
+    func defineLayoutForViews() {
+        gradientView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+
+        mainView.snp.makeConstraints {
+            $0.edges.equalTo(view.safeAreaLayoutGuide)
+        }
+
+        playerLabel.snp.makeConstraints {
+            $0.leading.top.equalToSuperview().inset(20)
+        }
+
+        pointsLabel.snp.makeConstraints {
+            $0.leading.greaterThanOrEqualTo(playerLabel.snp.trailing)
+            $0.trailing.top.equalToSuperview().inset(20)
+            $0.bottom.equalTo(playerLabel)
+        }
+
+        tableView.snp.makeConstraints {
+            $0.top.equalTo(playerLabel.snp.bottom).offset(5)
+            $0.leading.trailing.bottom.equalToSuperview()
+        }
+    }
+
+}
+
+extension LeaderboardViewController: UITableViewDataSource {
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        1
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        leaderboard.leaderboardPoints.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard
+            let cell = tableView.dequeueReusableCell(
+                withIdentifier: LeaderboardCell.reuseIdentifier,
+                for: indexPath) as? LeaderboardCell
+        else { return LeaderboardCell() }
+
+        let points = leaderboard.leaderboardPoints[indexPath.row]
+
+        cell.set(rank: indexPath.row, name: points.name, score: points.points)
```

Comment:
Remember that `indexPath.row` starts from zero, as seen in your video. Don't be hasty!